src/main/scala/exercise2/Ior2.scala

```diff
@@ -0,0 +1,59 @@
+package exercise2
+
+/**
+ * Ior[E, A] represents the inclusive or relationship, can contain E, A or both.
+ * map and flatMap will work on the right side (A).
+ * Only accept Throwable as E.
+ *
+ * @tparam A
+ */
+sealed trait Ior2[E <: Throwable, A] {
```

Comment:
Nicely done! `E` should actually not be restricted to `Throwable`, it can be anything. We just fixed it to that for the simpler task